com.hazelcast.jet.impl.execution.WatermarkCoalescer_TerminalSnapshotTest.test

https://hazelcast-l337.ci.cloudbees.com/job/jet/com.hazelcast.jet$hazelcast-jet-core/901/testReport/junit/com.hazelcast.jet.impl.execution/WatermarkCoalescer_TerminalSnapshotTest/test/

```
Error Message
expected:<RUNNING> but was:<COMPLETING>
Stacktrace
java.lang.AssertionError: expected:<RUNNING> but was:<COMPLETING>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.hazelcast.jet.impl.execution.WatermarkCoalescer_TerminalSnapshotTest.test(WatermarkCoalescer_TerminalSnapshotTest.java:136)
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/execution/WatermarkCoalescer_TerminalSnapshotTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/execution/WatermarkCoalescer_TerminalSnapshotTest.java
@@ -25,6 +25,7 @@ import com.hazelcast.jet.config.JetConfig;
 import com.hazelcast.jet.config.JobConfig;
 import com.hazelcast.jet.config.ProcessingGuarantee;
 import com.hazelcast.jet.core.JetTestSupport;
+import com.hazelcast.jet.core.JobStatus;
 import com.hazelcast.jet.datamodel.TimestampedEntry;
 import com.hazelcast.jet.pipeline.JournalInitialPosition;
 import com.hazelcast.jet.pipeline.Pipeline;
@@ -43,8 +44,10 @@ import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.LockSupport;
 
-import static com.hazelcast.jet.core.JobStatus.RUNNING;
-import static org.junit.Assert.assertEquals;
+import static com.hazelcast.jet.core.JobStatus.COMPLETED;
+import static com.hazelcast.jet.core.JobStatus.FAILED;
+import static com.hazelcast.jet.core.JobStatus.SUSPENDED;
+import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
 public class WatermarkCoalescer_TerminalSnapshotTest extends JetTestSupport {
@@ -74,7 +77,7 @@ public class WatermarkCoalescer_TerminalSnapshotTest extends JetTestSupport {
     @Test
     public void test() throws Exception {
         /*
-        This tests the issue that after a terminal barrier is processed, no other work should
+        This test tests the issue that after a terminal barrier is processed, no other work should
         be done by the ProcessorTasklet or CIES after that (except for emitting the DONE_ITEM).
         Also, if at-least-once guarantee is used, the tasklet should not continue to drain
         the queue that had the barrier.
@@ -133,7 +136,8 @@ public class WatermarkCoalescer_TerminalSnapshotTest extends JetTestSupport {
         }
 
         // check, if the job failed
-        assertEquals(RUNNING, job.getStatus());
+        JobStatus status = job.getStatus();
+        assertTrue(status != FAILED && status != COMPLETED && status != SUSPENDED);
     }
 
     private void producer(String key, int delayMs) {

